### ISSUE TITLE
Some entities do not have an accurate _stamp property

### ISSUE THREAD

**Issue by dstone760** [Wakanda/wakanda-javascript-client#66]
While working with the client, we noticed that our database had some entities that would not save. Upon further investigation, the issue was that the save() function was trying to save it as a new entity rather than an existing one. Upon even further investigation, I found this code in your entity_business.ts file, prepareDataForSave function: 
```typescript
if (this.entity._key && this.entity._stamp) {
      data.__KEY   = this.entity._key;
      data.__STAMP = this.entity._stamp;
} else {
      entityIsNew = true;
}
```
so, when I looked at those attributes on the entities that weren't saving for us, I found that those entities had a _stamp == 0, even though those entities were the result of a query so they could not possibly be 0. Our solution was to do a .fetch() on those entities and make changes to the result of that. 

It should be noted that not all entities had this problem, and our database has quite a history to it (at least 10 years). But the ones that did have that problem had it consistently. But I believe that part of the query should be whatever the fetch function does to the entity, and that this problem most likely has to do with legacy entities.

**Comment by fredGandelot** [comment]
Hi @dstone760 ,

Sorry for these simple questions, but i assume that you are using a 4D remote database ?
When checking via rest api or backend code, you can confirm me that those entities have a real __STAMP value  (and not 0 )

**Comment by dstone760** [comment]
Yes, we have a Wakanda DB and a 4D DB where we share data between the two, so we occasionally import data from the 4D DB to the Wakanda one. I meant to update this earlier, but while .fetch() worked for some entities, theres still some more that even after fetching, still had a _stamp of 0. My coworker found that these records were imported from 4D, but unfortunately I do not have many details of that import process, as I did not develop that. We can, however, overwrite the _stamp value on the clientside, save it, and then it retains that value.

**Comment by dstone760** [comment]
As a side note, I also cannot find much documentation on what the _stamp property actually is. So it's a little difficult to know how important this issue is, or how its coming about.

**Comment by fredGandelot** [comment]
Stamps allow optimistic locking management
This link should help you understanding how it works. 
http://doc.wakanda.org/home2.en.html#/Datastore/Entity/Locking-Entities.300-606099.en.html

However you can decide to overide the stamp mecanism. 

For now the most important thing would to know it the stamp value is equal to 0 (before import and after import) This way we would if the issue comes from you import function , or if the entities already have a stamp value equal to 0 in your 4D DB

**Comment by dstone760** [comment]
Well, the weird thing is that this problem only happens with the javascript client. When we do the normal prototyped pages, the entities save fine. So if its a problem with our database, I'm not sure how it wouldn't affect those as well.
